pkg/test/framework/components/echo/docker/workload.go

```diff
@@ -113,7 +113,7 @@ func newWorkload(e *native.Environment, cfg echo.Config, dumpDir string) (out *w
 	var env []string
 	var extraHosts []string
 	var capabilities []string
-	if cfg.Annotations.GetBool(echo.SidecarInject) {
+	if cfg.Workloads[0].Annotations.GetBool(echo.SidecarInject) {
```

Comment:
I think this is wrong ... the workload has to somehow know which config its for it.